.github/workflows/lint-and-test.yml

```diff
@@ -0,0 +1,33 @@
+name: Lint-and-test
+on: [pull_request]
+jobs:
+  call-workflow:
+    uses: ISISComputingGroup/reusable-workflows/.github/workflows/linters.yml@main
+    with:
+      compare-branch: origin/master
+      runs-on: windows-latest
+      python-ver: '3.11'
+      requirements-path: requirements-dev.txt
+  tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ "windows-latest" ]
+        version: ['3.11']
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.version }}
+      - name: install requirements
+        run: pip install -r requirements-dev.txt
+        env:
+          PIP_CONSTRAINT: "constraints.txt"
+      - name: run tests (linux)
+        run: python -m pytest . --cov
+        if: startsWith(matrix.os,'ubuntu')
```

Comment:
Probably don't need this as I don't think the build matrix has linux

Reply:
considering binning the matrix - we only test against windows-py311 at the moment